snapcraft/project/_sanity_checks.py

```diff
@@ -45,6 +47,27 @@ def conduct_project_sanity_check(project: snapcraft.project.Project):
     if os.path.isdir(snap_dir_path):
         _check_snap_dir(snap_dir_path)
 
+    # Unit tests are not seting the base, so we check for that here.
+    if project.info is not None:
+        _migration_notice(project.info.base)
+
+
+def _migration_notice(base: Optional[str]) -> None:
+    # We only want to encourage developers to use bases if using the snapcraft snap.
+    if base is None and common.is_snap():
+        logger.warning(
+            "This snapcraft project does not use bases which means this project "
+            "does not take advantage of the latest snapcraft features.\n"
+            "Read more about bases at https://docs.snapcraft.io/t/base-snaps/11198"
+        )
+    # Or if they are on the deb (this legacy codebase) and using bases.
+    elif base is not None and not common.is_snap():
+        logger.warning(
```

Comment:
Can we document in a comment what broadly are the differences between base support in the deb and in the snap? I can imagine revisiting this part of the code when someone is using bases with the deb, it breaks, and we want to know why.

Reply:
Bases in the deb is something only few Canonical folks used. I am fine with expanding the warning, but would rather this never make it to documentation.